### PR TITLE
Fix TreeItem shows cell edit in the wrong column when `select_mode=Row` and TreeItem has multiple columns

### LINKED ISSUE #88892: TreeItem shows cell edit in wrong column when SelectMode=Row
### Tested versions

Reproducible in:
- `v4.3.dev3.official [36e943b6b]`
- `v4.2.1.stable.official [b09f79f5]`
- `v4.1.4.rc1.official [b9008f3d5]`
- `v4.0.alpha1.official [31a7ddbf8]`
- `v3.5.3.stable.official [6c814135b]`
- `v3.3.stable.official`
- `v3.2.stable.official`

Unable to test in lower versions due to other bugs. Did not test in major versions lower than 3.

### System information

Godot v4.2.1.stable - Windows 10.0.19045 - GLES3 (Compatibility) - NVIDIA GeForce RTX 3070 (NVIDIA; 31.0.15.4633) - AMD Ryzen 5 5600X 6-Core Processor (12 Threads)

### Issue description

Selecting the last cell so as to edit it, actually edits the first cell. Other cells are not editable, despite calling `set_editable(column_idx, true)`.

**Expected**

At the very least, the TextEdit should visually be in the column that's being edited.

Ideally, clicking anywhere in the row (i.e. instead of a particular column, as indicated by the Select Mode) should trigger the editable behavior. Short of that, clicking the editable column should trigger the editable behavior.

### Steps to reproduce

Given a Tree where...
- the Tree has at least 1 TreeItem child
- All columns in the TreeItem(s) are set to editable i.e. `child.set_editable(column_idx, true)`
- Tree.columns = 3 (anything >1 repros, but tested primarily using 3)
- Select Mode set to `Row`

When the user clicks any column except the last column...
Then nothing is editable.

When the user clicks the last column...
Then the last column switches to a TextEdit containing the text in the *first column*.

When the user clicks the last column and edits the text...
The first column's text updates to the new text.

### Minimal reproduction project (MRP)

[TreeEditableBug.zip](https://github.com/godotengine/godot/files/14413993/TreeEditableBug.zip)

### PR DESCRIPTION
**Pull request by aqfranco**
Fix #88892

Editor would not be brought up when clicking on all but the last column on a TreeItem with SelectMode=Row with multiple columns and all editable, and when the editor was being brought up when clicking on the last column, it was editing the first column.

The reason for that was that the draw_item function was setting the set_meta parameter for that TreeItem for each loop, ignoring if it already had that parameter set, and the select_single_item function was only considering the first column when dealing with a TreeItem with SelectMode=Row.

Fixed draw_item, gui_input, and edit_selected functions by setting set_meta parameter for each column.
Fixed select_single_item function by changing selected_col depending on input.

### CODE REVIEW

File: scene/gui/tree.cpp [hunk 1]
```diff
@@ -3986,8 +3994,12 @@ bool Tree::edit_selected(bool p_force_edit) {
 	if (!s->cells[col].editable && !p_force_edit) {
 		return false;
 	}
-
-	Rect2 rect = s->get_meta("__focus_rect");
+	Rect2 rect;
+	if (select_mode == SELECT_ROW) {
+		rect = s->get_meta("__focus_col_" + char(selected_col));
```
Comment (AThousandShips) [inline scene/gui/tree.cpp:3999]:
```suggestion
		rect = s->get_meta("__focus_col_" + itos(selected_col));
```

File: scene/gui/tree.cpp [hunk 2]
```diff
@@ -3986,8 +3994,12 @@ bool Tree::edit_selected(bool p_force_edit) {
 	if (!s->cells[col].editable && !p_force_edit) {
 		return false;
 	}
-
-	Rect2 rect = s->get_meta("__focus_rect");
+	Rect2 rect;
```
Comment (AThousandShips) [inline scene/gui/tree.cpp:3997]:
```suggestion

	Rect2 rect;
```

**Comment by aqfranco** [comment]
Thanks for the suggestions, already committed the changes.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**